### ISSUE TITLE
Incompatibility with TypeScript 3.7.2

### ISSUE THREAD

**Issue by hjkcai** [fastly/performance-observer-polyfill#4]
The newly released TypeScript has added `supportedEntryTypes ` ([source](https://github.com/microsoft/TypeScript/blob/master/lib/lib.webworker.d.ts#L2569)) property to the `PerformanceObserver` instance. But it does not exist in @fastly/performance-observer-polyfill. This will lead to a compilation error.

Currently I have to use `skipLibCheck` to ignore this error. Please support the new version. Thanks very much!

**Comment by phamann** [comment]
Hi @hjkcai thank you for reporting this issue. 

I have tried to reproduce the error on a branch with TypeScript 3.7.2 [`phamann/update-dependencies`](https://github.com/fastly/performance-observer-polyfill/tree/phamann/update-dependencies) and the tests all pass successfully and the project builds. I have also imported the library into another project that uses it on 3.7.2 and it compiles without error.
 
Therefore, in order for me to assist, can you please give more details of the error you are seeing including the output from `tsc` and your TypeScript configuration.

**Comment by hjkcai** [comment]
@phamann Sorry that I didn't make myself clear. I was too busy and the issue was opened in hurry.

The problem appears when @fastly/performance-observer-polyfill is used as a dependency, not compiling the lib itself. Here is the minimum reproduction using TypeScript 3.7.2:

index.ts:

```typescript
import "@fastly/performance-observer-polyfill/polyfill";
```

tsconfig.json:

```json
{
  "compilerOptions": {
    "target": "es2015",
    "moduleResolution": "node",
    "strict": true
  },
  "include": [
    "index.ts"
  ]
}
```

The error message:

```
$ npx tsc -p .
node_modules/@fastly/performance-observer-polyfill/polyfill/index.ts:3:31 - error TS2322: Type '{ new (callback: PerformanceObserverCallback): PerformanceObserver; prototype: PerformanceObserver; readonly supportedEntryTypes: readonly string[]; } | typeof PerformanceObserver' is not assignable to type '{ new (callback: PerformanceObserverCallback): PerformanceObserver; prototype: PerformanceObserver; readonly supportedEntryTypes: readonly string[]; }'.
  Property 'supportedEntryTypes' is missing in type 'typeof PerformanceObserver' but required in type '{ new (callback: PerformanceObserverCallback): PerformanceObserver; prototype: PerformanceObserver; readonly supportedEntryTypes: readonly string[]; }'.

3 if (!ctx.PerformanceObserver) ctx.PerformanceObserver = PerformanceObserver;
                                ~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:11778:14
    11778     readonly supportedEntryTypes: ReadonlyArray<string>;
                       ~~~~~~~~~~~~~~~~~~~
    'supportedEntryTypes' is declared here.


Found 1 error.
```

Thanks!

**Comment by phamann** [comment]
@hjkcai Thanks for the detailed update I was able to repro with your suggestions and have now patched in https://github.com/fastly/performance-observer-polyfill/pull/6. Once thats reviewed and published to npm I'll comment back here with the new version number. Thanks again for the issue.

**Comment by phamann** [comment]
This has now been released in v1.1.0
